client/ladder/action-creators.ts

```diff
@@ -13,17 +8,21 @@ import { fetchJson } from '../network/fetch'
 
 const LADDER_RANKINGS_CACHE_TIME_MS = 60 * 1000
 
+type MatchmakingTypeAndSeasonId = `${MatchmakingType}|${SeasonId | undefined}`
+
 const lastFetchTimeByMatchmakingTypeAndSeasonId = new Map<MatchmakingTypeAndSeasonId, number>()
-const lastSearchTimeByMatchmakingTypeAndSeasonId = new Map<MatchmakingTypeAndSeasonId, number>()
-const lastSearchQueryByMatchmakingTypeAndSeasonId = new Map<MatchmakingTypeAndSeasonId, string>()
+const lastSearchInfoByMatchmakingTypeAndSeasonId = new Map<
+  MatchmakingTypeAndSeasonId,
+  { fetchTime: number; searchQuery: string }
+>()
 
 export function getCurrentSeasonRankings(
   matchmakingType: MatchmakingType,
   spec: RequestHandlingSpec<void>,
 ): ThunkAction {
   return abortableThunk(spec, async dispatch => {
     const fetchTime = performance.now()
-    const cacheKey = makeMatchmakingTypeAndSeasonId(matchmakingType)
+    const cacheKey = `${matchmakingType}|undefined` as MatchmakingTypeAndSeasonId
```

Comment:
Does this work if you do `cacheKey: MatchmakingTypeAndSeasonId = ...` instead? Would prefer that, as I worry the cast may not actually require the string to be in that form. (Apply throughout if so)